drivers/hmis/spec/requests/hmis/delete_client_spec.rb

```diff
@@ -0,0 +1,120 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'rails_helper'
+require_relative 'login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+RSpec.describe Hmis::GraphqlController, type: :request do
+  include_context 'hmis base setup'
+  let(:c1) { create :hmis_hud_client, data_source: ds1, user: u1 }
+  let(:c2) { create :hmis_hud_client, data_source: ds1, user: u1 }
+  let(:c3) { create :hmis_hud_client, data_source: ds1, user: u1 }
+  let!(:e1) { create :hmis_hud_enrollment, data_source: ds1, project: p1, client: c1, relationship_to_ho_h: 1, household_id: '1', user: u1 }
+
+  before(:each) do
+    hmis_login(user)
+    assign_viewable(edit_access_group, p1.as_warehouse, hmis_user)
+  end
+
+  let(:mutation) do
+    <<~GRAPHQL
+      mutation DeleteClient($input: DeleteClientInput!) {
+        deleteClient(input: $input) {
+          client {
+            id
+          }
+          #{error_fields}
+        }
+      }
+    GRAPHQL
+  end
+
+  def mutate(**kwargs)
+    response, result = post_graphql(**kwargs) { mutation }
+
+    aggregate_failures 'checking response' do
+      expect(response.status).to eq 200
+      client = result.dig('data', 'deleteClient', 'client')
+      errors = result.dig('data', 'deleteClient', 'errors')
+      yield client, errors
+    end
+  end
+
+  it 'should delete a client who has no multi-person enrollments' do
+    expect(Hmis::Hud::Client.all).to include(have_attributes(id: c1.id))
+
+    mutate(input: { id: c1.id }) do |client, errors|
+      expect(client).to be_present
+      expect(errors).to be_empty
+      expect(Hmis::Hud::Client.all).not_to include(have_attributes(id: c1.id))
+    end
+  end
+
+  it 'should delete a client who has a 2-person enrollment' do
+    e2 = create :hmis_hud_enrollment, data_source: ds1, project: p1, client: c2, relationship_to_ho_h: 2, household_id: '1', user: u1
+
+    mutate(input: { id: c1.id }) do |client, errors|
+      expect(client).to be_present
+      expect(errors).to be_empty
+      expect(Hmis::Hud::Client.all).to contain_exactly(have_attributes(id: c2.id))
+      expect(e2.reload.relationship_to_ho_h).to eq(1)
+    end
+  end
+
+  it 'should warn when trying to delete a client who has a 3+ person enrollment' do
+    e2 = create :hmis_hud_enrollment, data_source: ds1, project: p1, client: c2, relationship_to_ho_h: 2, household_id: '1', user: u1
+    e3 = create :hmis_hud_enrollment, data_source: ds1, project: p1, client: c3, relationship_to_ho_h: 2, household_id: '1', user: u1
+
+    mutate(input: { id: c1.id }) do |client, errors|
+      expect(client).to be_nil
+      expect(errors).to contain_exactly(
+        include(
+          'type' => 'information',
+          'severity' => 'warning',
```

Comment:
This is probably sufficient since it tests that the data is present. But just FYI on a pattern you could use for validating the actual message:
https://github.com/greenriver/hmis-warehouse/blob/75762fc75898fbd5f31965911fec42171d7087eb/drivers/hmis/spec/requests/hmis/update_relationship_to_ho_h_spec.rb#L112